### PR TITLE
Switch from harp to gulp for static build & deploy

### PR DESCRIPTION
**Pull request by lmorchard**
- Moved the static iframe content to `src/`
- Added a first attempt at an in-browser iframe harness to `src-harness/`
- New `gulpfile.js` and `gulpconfig.js` to handle static build, local dev server, and S3 deployment

### CODE REVIEW

**Comment by lmorchard** [comment]
So, this might be a tad over-engineered. But basically: This supports a static app with layers, so that you can deploy a production base app without the dev-mode harness I'm hacking on.

**Comment by lmorchard** [comment]
Two next steps that occur to me from this:
- Build and auto-deploy to S3 from master branch pushes via Travis CI
- Build and "deploy" the content into a local Firefox source tree clone to easily update the eventual pre-cached version of this content we'll need to ship with the browser.

File: src/scripts/adapters.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ searchSuggestionsAdapter = (function () {
 
   instance.messageReceived = function (e) {
     var message = e.detail.message;
+    if (!message) { return; }
```
Comment (jaredhirsch) [inline src/scripts/adapters.js:10]: nice, would also be nice to refactor these to one common superclass eventually

**Comment by jaredhirsch** [comment]
lgtm, let's do this!

**Pull request merged**